sourcegraph/sourcegraph
# app: Add new "Limited Access" banners

Design: [UX: Messaging/Feedback in App](https://www.figma.com/file/tbM21yr4k9gJNKkjNc0a4m/UX%3A-Messaging%2FFeedback-in-App?node-id=3%3A557&t=DPF8UOMNsPV0IWvv-0)

Adds a "Limited Access" banner on selected pages, explaining feature limitations in Sourcegraph App.

The banners only appear in Sourcegraph App mode.

- [x] Add new component `<LimitedAccessBanner>`
- [x] Add to Code Insights page
- [x] Add to Batch Changes page
- [x] Add to Notebooks page
- [x] Add CTA links that go to the right place
- [x] Make them dismissable via temporary settings

## Screenshots

<img width="1185" alt="Screen Shot 2023-03-03 at 1 51 37 PM" src="https://user-images.githubusercontent.com/602886/222803119-dfa74814-e975-49cd-87b7-5e344cc1969f.png">

<img width="1185" alt="Screen Shot 2023-03-03 at 1 48 12 PM" src="https://user-images.githubusercontent.com/602886/222802895-0bd91625-499b-4fc8-a31b-2899a233b0b2.png">

<img width="1185" alt="Screen Shot 2023-03-03 at 1 48 02 PM" src="https://user-images.githubusercontent.com/602886/222803013-8ee911bd-4aa3-44b0-9918-8f15166ecd89.png">


## Issues

This handles the following issues:

- #48395 
- #48393 
- #47569 (only the Limited Access banner in this PR)



## Test plan

- Run in Sourcegraph App mode
- Check presence of banners
- Banners are dismissable and stay dismissed (via temporary settings)
- Banner CTA links go to signup.sourcegraph.com with UTM query params
- Run in non-Sourcegraph App mode
- Banners do not appear

## App preview:

- [Web](https://sg-web-marek-add-limited-access-banner.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
